testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/realm/AddLdapRealmOnlineTest.java

```diff
@@ -0,0 +1,443 @@
+package org.wildfly.extras.creaper.commands.elytron.realm;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.commands.elytron.AbstractElytronOnlineTest;
+import org.wildfly.extras.creaper.commands.elytron.dircontext.AddDirContext;
+import org.wildfly.extras.creaper.core.CommandFailedException;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+
+@RunWith(Arquillian.class)
+public final class AddLdapRealmOnlineTest extends AbstractElytronOnlineTest {
+
+    private static final String TEST_LDAP_REALM_NAME = "CreaperTestLdapRealm";
+    private static final Address TEST_LDAP_REALM_ADDRESS = SUBSYSTEM_ADDRESS
+            .and("ldap-realm", TEST_LDAP_REALM_NAME);
+    private static final String TEST_LDAP_REALM_NAME2 = "CreaperTestLdapRealm2";
+    private static final Address TEST_LDAP_REALM_ADDRESS2 = SUBSYSTEM_ADDRESS
+            .and("ldap-realm", TEST_LDAP_REALM_NAME2);
+
+    private static final String TEST_DIR_CONTEXT_NAME = "CreaperTestDirContext";
+    private static final Address TEST_DIR_CONTEXT_ADDRESS = SUBSYSTEM_ADDRESS
+            .and("dir-context", TEST_DIR_CONTEXT_NAME);
+    private final AddDirContext addDirContext = new AddDirContext.Builder(TEST_DIR_CONTEXT_NAME)
+            .url("localhost")
+            .authenticationLevel(AddDirContext.AuthenticationLevel.NONE)
+            .build();
+
+    @After
+    public void cleanup() throws Exception {
+        ops.removeIfExists(TEST_LDAP_REALM_ADDRESS);
+        ops.removeIfExists(TEST_LDAP_REALM_ADDRESS2);
+        ops.removeIfExists(TEST_DIR_CONTEXT_ADDRESS);
+        administration.reloadIfRequired();
+    }
+
+    @Test
+    public void addSimpleLdapRealm() throws Exception {
+        client.apply(addDirContext);
+
+        AddLdapRealm addLdapRealm = new AddLdapRealm.Builder(TEST_LDAP_REALM_NAME)
+                .dirContext(TEST_DIR_CONTEXT_NAME)
+                .identityMapping(new AddLdapRealm.IdentityMappingBuilder()
+                        .rdnIdentifier("someId")
+                        .userPasswordMapper(new AddLdapRealm.UserPasswordMapperBuilder()
```

Comment:
It is currently required due to https://issues.jboss.org/browse/JBEAP-6301 - however PR for this issue has been already sent. Once it will appear in EAP then this userPasswordMapper will be removed from simple commands.